streams/src/test/java/org/apache/kafka/streams/processor/internals/RepartitionWithMergeOptimizingTest.java

```diff
@@ -192,104 +211,103 @@ private int getCountOfRepartitionTopicsFound(final String topologyString) {
         return keyValueList;
     }
 
-
-
     private static final String EXPECTED_OPTIMIZED_TOPOLOGY = "Topologies:\n"
                                                               + "   Sub-topology: 0\n"
-                                                              + "    Source: KSTREAM-SOURCE-0000000000 (topics: [inputA])\n"
-                                                              + "      --> KSTREAM-MAP-0000000002\n"
-                                                              + "    Source: KSTREAM-SOURCE-0000000001 (topics: [inputB])\n"
-                                                              + "      --> KSTREAM-MAP-0000000003\n"
-                                                              + "    Processor: KSTREAM-MAP-0000000002 (stores: [])\n"
-                                                              + "      --> KSTREAM-MERGE-0000000004\n"
-                                                              + "      <-- KSTREAM-SOURCE-0000000000\n"
-                                                              + "    Processor: KSTREAM-MAP-0000000003 (stores: [])\n"
-                                                              + "      --> KSTREAM-MERGE-0000000004\n"
-                                                              + "      <-- KSTREAM-SOURCE-0000000001\n"
-                                                              + "    Processor: KSTREAM-MERGE-0000000004 (stores: [])\n"
-                                                              + "      --> KSTREAM-FILTER-0000000021\n"
-                                                              + "      <-- KSTREAM-MAP-0000000002, KSTREAM-MAP-0000000003\n"
-                                                              + "    Processor: KSTREAM-FILTER-0000000021 (stores: [])\n"
-                                                              + "      --> KSTREAM-SINK-0000000020\n"
-                                                              + "      <-- KSTREAM-MERGE-0000000004\n"
-                                                              + "    Sink: KSTREAM-SINK-0000000020 (topic: KSTREAM-AGGREGATE-STATE-STORE-0000000005-repartition)\n"
-                                                              + "      <-- KSTREAM-FILTER-0000000021\n"
+                                                              + "    Source: KSTREAM-SOURCE-0000000020 (topics: [long-groupByKey-repartition])\n"
```

Comment:
Hmm.. why the sub-topologies ordering was reversed here?

Reply:
@guozhangwang I think this has to do with the fact that `TopologyDescription` sorts the topology strings

Reply:
Yeah I know, but the SubtopologyComparator sorts by the sub-topology-id here. What surprises me is that, those two sub-topologies actually switched their ids too -- usually we assign the ids to sub-topologies following the parsing order, right?

Reply:
That's what I thought as well. But `makeNodeGroups` sorts by name first then iterates over the names and assigns sub-topology id. Which I think would explain it.

Reply:
Ah you're right.

Hmm I think we should fix the logic

```
for (final String nodeName : Utils.sorted(allSourceNodes)) {
            nodeGroupId = putNodeGroupName(nodeName, nodeGroupId, nodeGroups, rootToNodeGroup);
        }
```

here asap, to sort by the sink -> source relationship than by source node names; previously we do not expect the source node to be customized normally so that we can just rely on the internal suffix which is basically following the parsing order. But we do not need to get this fix in this PR.

Reply:
I agree, I found it really confusing that the subtopologies were suddenly in a different, and seemingly incorrect, order (thanks for filing a ticket)